overlord/snapshotstate/backend/backend.go

```diff
@@ -323,3 +326,97 @@ func addDirToZip(ctx context.Context, snapshot *client.Snapshot, w *zip.Writer,
 
 	return nil
 }
+
+type exportMetadata struct {
+	Format int       `json:"format"`
+	Date   time.Time `json:"date"`
+}
+
+// Export allows exporting a given snapshot set to the given writer.
+func Export(ctx context.Context, setID uint64, w io.Writer) error {
+	// Open all files first and keep the file descriptors open.
+	//
+	// This ensures that even if a snapshot is removed we can
+	// still serve the data. Hence no need for locking.
```

Comment:
nice!

Reply:
this is clever, but also means in a corner case this can produce instead of an error a partial snapshot. I think  we need a compromise where we have a PrepareExport that needs to be called holding the state lock. And then it returns something that has just the opened file with a method to do the actual export...

Reply:
This is done now.